clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/EcsControllersSpec.java

```diff
@@ -18,26 +18,95 @@
+import java.util.*;
 import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
 import org.springframework.beans.factory.annotation.Autowired;
 
 public class EcsControllersSpec extends EcsSpec {
 
   @Autowired private ProviderRegistry providerRegistry;
+  private AmazonECS mockECS = mock(AmazonECS.class);
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given cached ECS clusters (names), retrieve detailed description "
+          + "of the cluster from /ecs/ecsDescribeClusters/{account}/{region}"
+          + "\n===")
+  @Test
+  public void getAllEcsClusterDetailsTest() throws JsonProcessingException {
+    // given
+    ProviderCache ecsCache = providerRegistry.getProviderCache(EcsProvider.NAME);
+    String testClusterName = "example-app-test-Cluster-NSnYsTXmCfV2";
+    String testNamespace = Keys.Namespace.ECS_CLUSTERS.ns;
+
+    String clusterKey = Keys.getClusterKey(ECS_ACCOUNT_NAME, TEST_REGION, testClusterName);
+    Map<String, Object> attributes = new HashMap<>();
+    attributes.put("account", ECS_ACCOUNT_NAME);
+    attributes.put("region", TEST_REGION);
+    attributes.put("clusterArn", "arn:aws:ecs:::cluster/" + testClusterName);
+    attributes.put("clusterName", testClusterName);
+
+    DefaultCacheResult testResult = buildCacheResult(attributes, testNamespace, clusterKey);
+    ecsCache.addCacheResult("TestAgent", Collections.singletonList(testNamespace), testResult);
+
+    when(mockAwsProvider.getAmazonEcs(any(NetflixECSCredentials.class), anyString(), anyBoolean()))
+        .thenReturn(mockECS);
+
+    Cluster clusterDecription =
+        new Cluster()
+            .withClusterArn("arn:aws:ecs:::cluster/" + testClusterName)
+            .withStatus("ACTIVE")
+            .withCapacityProviders("FARGATE", "FARGATE_SPOT")
+            .withClusterName(testClusterName);
+    when(mockECS.describeClusters(any(DescribeClustersRequest.class)))
+        .thenReturn(new DescribeClustersResult().withClusters(clusterDecription));
+
+    // when
+    String testUrl =
+        getTestUrl("/ecs/getEcsClusterDescriptions/" + ECS_ACCOUNT_NAME + "/" + TEST_REGION);
+
+    Response response =
+        get(testUrl).then().statusCode(200).contentType(ContentType.JSON).extract().response();
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    Collection<Cluster> clusters =
+        Arrays.asList(objectMapper.readValue(response.asString(), Cluster[].class));
+    // then
+    assertNotNull(clusters);
+    Cluster clusterDescription =
+        (clusters.stream().filter(cluster -> cluster.getClusterName().equals(testClusterName)))
+            .findAny()
+            .get();
+    assertTrue(clusterDescription.getClusterArn().contains(testClusterName));
+    assertTrue(clusterDescription.getCapacityProviders().size() == 2);
```

Comment:
nit: Since you're comparing 2 values, you might want to use assertEquals here (it provides the user with a better error "expected * actual *")